File: leetcode/201-300/231-power-of-two/bench/power_of_two.c
#include <stdio.h>

static int is_power_of_two(long n) {
    if (n <= 0) return 0;
    return (n & (n - 1)) == 0;
}

int main(void) {
    long n = 130000000;
    long mask = 1023;
    long state = 12345;
    long sink = 0;
    for (long i = 0; i < n; i++) {
        state = (state * 1103515245L + 12345L) & 2147483647L;
        long v = state & mask;
        if (is_power_of_two(v)) sink++;
    }
    printf("%ld\n", sink);
    return 0;
}

File: leetcode/201-300/231-power-of-two/bench/power_of_two.py
"""Benchmark workload for LeetCode #231 — Power of Two (Python; scale lane)."""


def is_power_of_two(n):
    if n <= 0:
        return False
    return (n & (n - 1)) == 0


def main():
    n = 130000000
    mask = 1023
    state = 12345
    sink = 0
    for _ in range(n):
        state = (state * 1103515245 + 12345) & 2147483647
        v = state & mask
        if is_power_of_two(v):
            sink += 1
    print(sink)


main()
